manifests/vhost.pp

```diff
@@ -2707,7 +2711,64 @@
   # - $ssl_openssl_conf_cmd
   # - $ssl_stapling
   # - $apache_version
-  if $ssl {
+  if $ssl and $ensure == 'present' {
+    if $ssl_reload_on_change {
+      if $ssl_cert {
+        include apache::mod::ssl::reload
+        $_ssl_cert_copy = regsubst($ssl_cert, '/', '_', 'G')
+        file { "${filename}${_ssl_cert_copy}":
```

Comment:
There's a lot of duplication for this. Can't this be captured in a define somehow and reused? Perhaps a loop?

```puppet
[$ssl_cert, $ssl_key, $ssl_ca, $ssl_chain, $ssl_crl].each |$ssl_file| {
  if $ssl_file {
    include apache::mod::ssl::reload
    $_ssl_file_copy = regsubst($ssl_file, '/', '_', 'G')
    file { "${filename}${_ssl_file_copy}":
      path    => "${apache::params::puppet_ssl_dir}/${filename}${_ssl_file_copy}",
      source  => "file://${ssl_file}",
      mode    => '0640',
      seltype => 'cert_t',
      notify  => Class['apache::service'],
    }
  }
} 
```

Reply:
Thats actually a good idea, will implement that.

Reply:
Did the same in apache::mod::ssl